taui/src/components/anonymous-authenticator.js

```diff
@@ -0,0 +1,72 @@
+// @flow
+import { Component, Fragment } from 'react'
+import LogRocket from 'logrocket'
+
+import {ANONYMOUS_USERNAME, PROFILE_CONFIG_KEY} from '../constants'
+import {storeConfig} from '../config'
+
+import CustomSignIn from './custom-sign-in'
+import CustomHeaderBar from './custom-header-bar'
+
+export default function anonymousAuthenticator (Comp) {
+  return class extends Component {
+    constructor (props) {
+      super(props)
+
+      this.handleAnonymousLogin = this.handleAnonymousLogin.bind(this)
+      this.handleProfileChange = this.handleProfileChange.bind(this)
+
+      // Load the selected user profile from localStorage, if any
+      this.props.loadProfile()
+    }
+
+    handleProfileChange (profile: AccountProfile) {
+      storeConfig(PROFILE_CONFIG_KEY, profile)
+      this.props.store.dispatch({type: 'set profile', payload: profile})
```

Comment:
```suggestion
      this.props.setProfile(profile)
```
This prop calls both and has some error handling